src/components/VUETable.tsx

```diff
@@ -1,44 +1,73 @@
-import React from 'react';
-import { default as VUEData } from './../data/VUEs.json';
+import React, { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import { VUE } from '../model/VUE';
+import { fetchVueData } from '../utils/VUEDataFetcherUtils';
+import "./VUETable.css";
 
 import vueLogo from "./../images/vue_logo.png";
 
-class VUETable extends React.Component<{}> {
-    public render() {
-        const DisplayData = VUEData.map((info) => {
-            return (
-                <tr>
-                    <td>{info.hugoGeneSymbol}</td>
-                    <td>{info.genomicLocation}</td>
-                    <td>{info.defaultEffect}</td>
-                    <td>{info.comment}</td>
-                    <td>{info.context}{info.referenceText && (<>{' '}<a href={`https://pubmed.ncbi.nlm.nih.gov/${info.pubmedIds[0]}/`} rel="noreferrer" target="_blank">({info.referenceText})</a></>)}</td>
-                    <td>{info.revisedProteinEffects && (<a href={`https://deploy-preview-139--genome-nexus-frontend.netlify.app/variant/${info.revisedProteinEffects[0].variant}`} rel="noreferrer" target="_blank">Genome Nexus <i className="fa fa-external-link" /></a>)}</td>
-                </tr>
-            );
-        });
+export const VUETable: React.FC = () => {
+    const [vueData, setVueData] = useState<VUE[]>([]);
+
+    useEffect(() => {
+        const fetchData = async () => {
+          const data = await fetchVueData();
+          setVueData(data);
+        };
+
+        fetchData();
+    }, []);
+
+    const DisplayData = vueData.map((info) => {
+        const references: { referenceText: string, pubmedId: number }[] = [];
+        info.revisedProteinEffects.map(v => ({ referenceText: v.referenceText, pubmedId: v.pubmedId })).filter(r => {
+            let i = references.findIndex(ref => (ref.referenceText === r.referenceText && ref.pubmedId === r.pubmedId))
+            if (i === -1) {
+                references.push(r)
+            }
+            return null;
+        })
+
+        const links = references.map<React.ReactNode>(reference => 
+            (<>
+                <a href={`https://pubmed.ncbi.nlm.nih.gov/${reference.pubmedId}/`} rel="noreferrer" target="_blank">
+                    ({reference.referenceText})
+                </a>
+            </>)
+        ).reduce((prev, curr) => [prev, ', ', curr])
```

Comment:
```suggestion
        ).reduce((prev, curr) => [prev, ', ', curr]);
```

Reply:
Not sure if this could work, is it going to create a nested array, e.g. Array [Array [Array [1, ", ", 2], ", ", 3], ", ", 4]?